### PR TITLE
ui: Create Routing Configurations route and page

### PR DESCRIPTION
**Pull request by kaxcode**
### ✨ Description:

This PR creates the Routing Configurations route and page.

Why: A routing configuration is not a real service. They are a routing configuration that route traffic to real services in Consul. For this reason we created a new route for routing-configs that does not follow the services `/dc/service/:name-of-service` pattern.

Changes:

- Created the new route `dc/routing-config/:name` and template for Routing Configurations. The Routing Config page will only show Discovery Chain. No other tabs/pages are expected to be added in the future.
- Create `Consul::Source` component. This is a component very similar to `Consul::Kind` . Consul uses the attribute Kind to describe Services, e.g. Terminating Gateway, Mesh Gateway, etc. Routing Configuration are not Services and so I create a new pill component to be used by any item by does not relate to Services.

[Demo Link](https://consul-ui-staging-4w5nzpsql-hashicorp.vercel.app/ui/dc1/routing-config/virtual-1#CONSUL_ACLS_ENABLE=1
)

### 📸 Screenshots:
Screen Shot 2021-08-12 at 9 42 08 AM

### ⚡ Backend Changes:

No backend changes required.

### 🤡 Updates to mock-api:

No updates to mock data.

### 🧪 Testing:

Mostly presentational tests. Tests cover the following...

- The URL redirects to the correct template by checking page title
- When redirected you don't get a 404 error because the URL exists
- Check that the new `Consul::Source` component is there. It should always be there.

### 📝  PR Tasks

- [x] Changelog
- [x] Demo Link

### CODE REVIEW

File: ui/packages/consul-ui/app/components/consul/source/README.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+```hbs preview-template
+<Consul::Source @source='Source Name' />
+```
+
+### Arguments
+
+| Argument/Attribute | Type | Default | Description |
+| --- | --- | --- | --- |
+| `source` | `string` |  | A string to be passed down and displayed as the source name. |
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/components/consul/source/README.mdx:13]: I went with string here instead of an object because to make this component more generic. This is easier than scoping all the possible objects that could be received.

File: ui/packages/consul-ui/app/styles/layout.scss [hunk 2]
```diff
@@ -49,7 +49,8 @@ html[data-route$='edit'] .app-view > header + div > *:first-child {
 /* needs a top margun :S */
 %app-view-content .tab-section > *:first-child:not(.filter-bar):not(table),
 %app-view-content .tab-section > .search-bar + p,
-%app-view-content .tab-section .consul-health-check-list {
+%app-view-content .tab-section .consul-health-check-list,
+%app-view-content .container {
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/styles/layout.scss:53]: The Routing Config template does not have a `tab-nav`. I opted for using a generic class name to wrap around `Consul::DiscoveryChain` instead of using `tab-section` which wouldn't make sense.

File: ui/packages/consul-ui/app/templates/dc/routing-config.hbs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{{page-title @model.slug}}
+
+<AppView>
+  <BlockSlot @name="breadcrumbs">
+    <ol>
+        <li><a data-test-back href={{href-to 'dc.services'}}>All Services</a></li>
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/templates/dc/routing-config.hbs:6]: I thought about moving this to translations but I thought it would be best to have all breadcrumbs moved to translations in a separate PR.

File: ui/packages/consul-ui/translations/components/consul/en-us.yaml [hunk 4]
```diff
@@ -152,3 +152,9 @@ topology-metrics:
       header: No traffic
       body: Add "{upstream}" as an explicit upstream of "{downstream}" or set the "{downstream}" proxy to "transparent" mode to enable traffic.
       action: Documentation
+source:
+  header: This is not a registered Consul service. It’s a routing configuration that routes traffic to real services in Consul. For more information, read our documentation.
+  menu-title: About Routing Configs
+  links:
```
Comment (kaxcode) [inline ui/packages/consul-ui/translations/components/consul/en-us.yaml:158]: In the future we will add another link for the consul learn site.

**Review by randallmorey — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/427910.
